packages/lib/services/profile.ts

```diff
@@ -0,0 +1,81 @@
+import { prisma } from "@formbricks/database";
+import { DatabaseError, ResourceNotFoundError } from "@formbricks/errors";
+import { Prisma } from "@prisma/client";
+import { TProfile, TProfileUpdateOutput } from "@formbricks/types/v1/profile";
+import { cache } from "react";
+
+const responseSelection = {
+  id: true,
+  name: true,
+  email: true,
+};
+
+// function to retrive basic information about a user's profile
+export const getProfile = cache(async (userId: string): Promise<TProfile> => {
+  try {
+    const profile = await prisma.user.findUnique({
+      where: {
+        id: userId,
+      },
+      select: responseSelection,
+    });
+
+    if (!profile) {
+      throw new ResourceNotFoundError("Profile", userId);
```

Comment:
I'm currently changing the other services. turned out it is annoying to always catch the exception when a ressource it not found. returning null (like prisma does) is much easier to handle. So please just return the result from prisma and change the return type of this function to `Promise<TProfile|null>` 😊